xarray/core/dataset.py

```diff
@@ -1123,27 +1126,36 @@ def isel(self, drop=False, **indexers):
         See Also
         --------
         Dataset.sel
-        Dataset.sel_points
-        Dataset.isel_points
         DataArray.isel
         """
+        from .dataarray import DataArray
+
         invalid = [k for k in indexers if k not in self.dims]
         if invalid:
             raise ValueError("dimensions %r do not exist" % invalid)
 
-        # all indexers should be int, slice or np.ndarrays
-        indexers = [(k, (np.asarray(v)
-                         if not isinstance(v, integer_types + (slice,))
-                         else v))
+        # extract new coordinates from indexers
```

Comment:
Actually, this already happens in `expand_indexers`, so there's not really any point in adding alignment with `join='exact'`. Anyways I sent you https://github.com/fujiisoup/xarray/pull/2 with a few tweaks, none of which are terribly important.